dispatcher.go

```diff
@@ -134,11 +134,27 @@ func NewDispatcher(cfg Config) *Dispatcher {
 func addObservingMiddleware(cfg Config, logger *zap.Logger) Config {
 	observer := observerware.New(logger, observerware.NewNopContextExtractor())
 
-	cfg.InboundMiddleware.Unary = inboundmiddleware.UnaryChain(observer, cfg.InboundMiddleware.Unary)
```

Comment:
Makes me wonder if we can put this behavior directly into these Chain funcs, instead of at these callsites... what do you think @akshayjshah @kriskowal